### PR TITLE
[cargo] Downgrade bindgen to unyanked 0.54.0, upgrade others.  clang-sys 1.0.0/0.23.0 collision prevented bindgen upgrade.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Dependabot see's our current version of bindgen is yanked and tries to upgrade bindgen for us.
It can't due to recent release of libclang-sys 1.0.0.

Rocksdb's bindgen 0.54.0 wants libclang-sys 0.23.0, and newer bindgen 0.55.0 wants libclang-sys 1.0.0.  Since there can be only one for linking...

Commands: 

cargo upgrade --workspace anyhow async-trait cc chrono clap flate2 handlebars hmac kube libc mirai-annotations once_cell pbkdf2 proptest reqwest rustls serde structopt syn trybuild rayon ureq num-derive serde-generate bindgen@0.54.0

cargo update

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

CI

## Related PRs

None

### CODE REVIEW

**Comment by thefallentree** [comment]
not sure why i am an code owner, but this upgrade looks okay to me

**Comment by bmwill** [comment]
@rexhoffman We should try to avoid doing batch upgrades like this, at the very least we should have commits for each individual upgrade (or small groups of upgrades if needed)

**Comment by rexhoffman** [comment]
@bmwill I'm thinking of turning back on dependabot, until a proper date is set - and we're a month out.

File: client/libra-dev/Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ transaction-builder = { path = "../../language/transaction-builder", version = "
 rand = "0.7.3"
 
 [build-dependencies]
-bindgen = "0.54.1"
```
Comment (sausagee) [inline client/libra-dev/Cargo.toml:27]: Just want to confirm this is intended.

File: .dependabot/config.yml [hunk 2]
```diff
@@ -9,6 +9,10 @@ update_configs:
           dependency_name: "codespan*"
       - match:
           dependency_name: "libfuzzer-sys*"
+      - match:
+          dependency_name: "bindgen*"
+      # Rocksdb depends on bindgen ^0.54.0 -- which depends on clang-sys ^0.23.0
+      # bindgen 0.55.0 depends on clang-sys 1.0.0, and since there can be only one for linking...
```
Comment (sausagee) [inline .dependabot/config.yml:15]: @lightmark Do we have plan to bump rocksdb?

File: client/libra-dev/Cargo.toml (hunk 1 above)
Comment (rexhoffman) [reply client/libra-dev/Cargo.toml:27]: yes it is, need to downgrade, upgrade is blocked by conflicts on clang-sys 1.0.0/0.23.0

**Comment by rexhoffman** [comment]
@bmwill @sunshowers 
So this is the compilation that breaks if rustls is upgrades.   

This is the command to make the change.
```
cargo upgrade --workspace rustls
```

This is the delta
```
libra-vault-client:
    Upgrading rustls v0.17.0 -> v0.18.1
```

```
   Compiling libradb v0.1.0 (/Users/rexhoffman/git/libra/storage/libradb)
error[E0308]: mismatched types
   --> secure/storage/vault/src/lib.rs:357:36
    |
357 |             request.set_tls_config(tls_config.clone());
    |                                    ^^^^^^^^^^^^^^^^^^ expected struct `rustls::client::ClientConfig`, found a different struct `rustls::client::ClientConfig`
    |
    = note: expected struct `std::sync::Arc<rustls::client::ClientConfig>` (struct `rustls::client::ClientConfig`)
               found struct `std::sync::Arc<rustls::client::ClientConfig>` (struct `rustls::client::ClientConfig`)
    = note: perhaps two different versions of crate `rustls` are being used?

   Compiling libra-vm v0.1.0 (/Users/rexhoffman/git/libra/language/libra-vm)
```

I suspect some form of coercion should be used here.

**Comment by rexhoffman** [comment]
/land

Before the lock files change again....

**Comment by rexhoffman** [comment]
really again?

**Pull request closed without merging**